### PR TITLE
Helm 2 0 0 release

### PR DESCRIPTION
**Pull request by bhagyashree-sarawate**
Hi Shiva
We have finalized the CSP image as hpestorage/hpe3parprimera-csp:v1.0.0 .Modified helm charts accordingly for the CSP image.Created the package using package.sh and added under docs.
Tested the deployment using this package.
Also removed the serviceName from 2.0.0 CSV file as per our discussion in morning.
Please review.

Output
=========
[root@k8s-master00 hpe-csi-driver]# helm install hpe-csi /root/bhagyashree/helm_2_0_0/co-deployments/docs/hpe-csi-driver-2.0.0.tgz -n kube-system -f values_3par.yaml
NAME: hpe-csi
LAST DEPLOYED: Thu Apr 16 16:09:31 2020
NAMESPACE: kube-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
[root@k8s-master00 hpe-csi-driver]# helm ls -n kube-system
NAME    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                   APP VERSION
hpe-csi kube-system     1               2020-04-16 16:09:31.19549352 +0530 IST  deployed        hpe-csi-driver-2.0.0    1.1.1

[root@k8s-master00 hpe-csi-driver]# kubectl get pods -n kube-system
NAME                                                         READY   STATUS    RESTARTS   AGE
hpe-csi-controller-84d8569476-kszl6                          5/5     Running   0          54s
hpe-csi-node-grzb2                                           2/2     Running   0          54s
hpe-csi-node-qj2qc                                           2/2     Running   0          54s
hpe3parprimera-csp-6847b9f649-4d29m                          1/1     Running   0          54s

[root@k8s-master00 hpe-csi-driver]# kubectl describe pod hpe3parprimera-csp-6847b9f649-4d29m -n kube-system |grep image
  Normal  Pulling    77s   kubelet, k8s-worker01.in.rdlabs.hpecorp.net  Pulling image "hpestorage/hpe3parprimera-csp:v1.0.0"
  Normal  Pulled     67s   kubelet, k8s-worker01.in.rdlabs.hpecorp.net  Successfully pulled image "hpestorage/hpe3parprimera-csp:v1.0.0"

[root@k8s-master00 hpe-csi-driver]# kubectl describe pod hpe-csi-controller-84d8569476-kszl6 -n kube-system |grep image
  Normal  Pulling    107s  kubelet, k8s-worker02.in.rdlabs.hpecorp.net  Pulling image "quay.io/k8scsi/csi-provisioner:v1.5.0"
  Normal  Pulled     96s   kubelet, k8s-worker02.in.rdlabs.hpecorp.net  Successfully pulled image "quay.io/k8scsi/csi-provisioner:v1.5.0"
  Normal  Pulling    96s   kubelet, k8s-worker02.in.rdlabs.hpecorp.net  Pulling image "quay.io/k8scsi/csi-attacher:v2.1.1"
  Normal  Pulled     87s   kubelet, k8s-worker02.in.rdlabs.hpecorp.net  Successfully pulled image "quay.io/k8scsi/csi-attacher:v2.1.1"
  Normal  Pulling    87s   kubelet, k8s-worker02.in.rdlabs.hpecorp.net  Pulling image "quay.io/k8scsi/csi-snapshotter:v2.0.1"
  Normal  Pulling    82s   kubelet, k8s-worker02.in.rdlabs.hpecorp.net  Pulling image "quay.io/k8scsi/csi-resizer:v0.4.0"
  Normal  Pulled     82s   kubelet, k8s-worker02.in.rdlabs.hpecorp.net  Successfully pulled image "quay.io/k8scsi/csi-snapshotter:v2.0.1"
  Normal  Pulled     71s   kubelet, k8s-worker02.in.rdlabs.hpecorp.net  Successfully pulled image "quay.io/k8scsi/csi-resizer:v0.4.0"
  Normal  Pulling    71s   kubelet, k8s-worker02.in.rdlabs.hpecorp.net  Pulling image "hpestorage/csi-driver:v1.1.1"
  Normal  Pulled     67s   kubelet, k8s-worker02.in.rdlabs.hpecorp.net  Successfully pulled image "hpestorage/csi-driver:v1.1.1"

Regards
Bhagyashree

### CODE REVIEW

File: helm/charts/hpe-csi-driver/Chart.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ sources:
 - https://github.com/hpe-storage/csi-driver
 name: hpe-csi-driver
 home: https://hpe.com/storage/containers
-version: 2.0.0-beta
+version: 2.0.0
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/Chart.yaml:12]: Why the huge version bump?

File: helm/charts/hpe-csi-driver/README.md [hunk 2]
```diff
@@ -36,8 +36,8 @@ The following table lists the configurable parameters of the HPE-CSI chart and t
 | storageClass.parameters.fsType                    | Type of file system being used (ext4, ext3, xfs, btrfs)     | xfs         |
 | storageClass.parameters.volumeDescription         | Volume description for volumes created using HPE CSI driver     | -         |
 | storageClass.parameters.accessProtocol            | Access protocol to use for storage connectivity (iscsi, fc)     | iscsi         |
-| storageClass.parameters.provisioningType         | Provisioning type for HPE 3PAR Primera     | tpvv         |
-| storageClass.parameters.cpg            | CPG type for HPE 3PAR Primera     | FC_r6         |
+| storageClass.parameters.provisioningType         | Provisioning type for HPE 3PAR Primera(tpvv, tdvv, full)     | tpvv         |
+| storageClass.parameters.cpg            |  Existing CPG name for HPE 3PAR Primera     | FC_r6         |
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md:40]: CSP specific `StorageClass` parameters should not leak out into generic parts. Please remove.

File: helm/charts/hpe-csi-driver/values.yaml [hunk 3]
```diff
@@ -41,9 +41,9 @@ storageClass:
     fsType: xfs
     volumeDescription: "Volume created by the HPE CSI Driver for Kubernetes"
     accessProtocol: "iscsi"
-    # Provisioning type for 3PAR primera
+    # Provisioning type for 3PAR primera (tpvv, tdvv, full)
     provisioningType: "tpvv"
-    # CPG for 3PAR primera
+    # Existing CPG name for 3PAR primera
     cpg: "FC_r6"
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/values.yaml:47]: Why are these parameters leaking out here when they have CSP defaults and are CSP specific?

File: operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml [hunk 4]
```diff
@@ -21,7 +21,6 @@ metadata:
               "backend": "192.168.1.1",
               "create": true,
               "password": "admin",
-              "serviceName": "nimble-csp-svc",
```
Comment (drajen) [inline operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml:24]: Please leave this in as "backend-csp-svc"

File: operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml [hunk 5]
```diff
@@ -147,11 +146,6 @@ spec:
             path: secret.password
             x-descriptors:
               - 'urn:alm:descriptor:com.tectonic.ui:password'
-          - displayName: CSP service hostname
-            description: CSP service hostname or IP address
-            path: secret.serviceName
-            x-descriptors:
-              - 'urn:alm:descriptor:com.tectonic.ui:text'
```
Comment (drajen) [inline operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml:154]: Why is this being removed?

File: operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml [hunk 6]
```diff
@@ -193,11 +187,11 @@ spec:
             x-descriptors:
               - 'urn:alm:descriptor:com.tectonic.ui:text'
           - displayName: Provisioning type for 3PAR Primera
-            description: 'Provisioning type for 3PAR Primera (default: tpvv choices: tpvv)'
+            description: 'Provisioning type for 3PAR Primera (default: tpvv choices: tpvv, tdvv, full)'
```
Comment (drajen) [inline operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml:190]: Delete

File: operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml [hunk 7]
```diff
@@ -193,11 +187,11 @@ spec:
...
             path: storageClass.parameters.provisioningType
             x-descriptors:
               - 'urn:alm:descriptor:com.tectonic.ui:text'
-          - displayName: CPG for 3PAR Primera
+          - displayName: Existing CPG for 3PAR Primera
```
Comment (drajen) [inline operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml:194]: Delete

File: helm/charts/hpe-csi-driver/README.md (hunk 2 above)
Comment (bhagyashree-sarawate) [reply helm/charts/hpe-csi-driver/README.md]: Removed

File: helm/charts/hpe-csi-driver/values.yaml (hunk 3 above)
Comment (bhagyashree-sarawate) [reply helm/charts/hpe-csi-driver/values.yaml]: removed

File: operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml (hunk 4 above)
Comment (bhagyashree-sarawate) [reply operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml:24]: This is handled internally in the templates depending on the backendtype ,so no part of values.yaml so removed.

File: operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml (hunk 5 above)
Comment (bhagyashree-sarawate) [reply operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml:154]: serviceName is handled internally by the code templates

File: operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml (hunk 6 above)
Comment (bhagyashree-sarawate) [reply operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml]: Removed, done.

File: operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml (hunk 7 above)
Comment (bhagyashree-sarawate) [reply operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml]: done

File: docs/index.yaml [hunk 8]
```diff
@@ -1,6 +1,29 @@
...
+    - RHEL
+    maintainers:
+    - email: hpe-containers-dev@hpe.com
+      name: shivamerla
+    name: hpe-csi-driver
+    sources:
+    - https://github.com/hpe-storage/csi-driver
+    urls:
+    - hpe-csi-driver-2.0.0.tgz
+    version: 2.0.0
```
Comment (shivamerla) [inline docs/index.yaml:26]: Can you change version to 1.2.0 and delete old tgz 2.0.0-beta.

File: operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml [hunk 9]
```diff
@@ -34,8 +33,6 @@ metadata:
                 "accessProtocol": "iscsi",
                 "fsType": "xfs",
                 "volumeDescription": "Volume created by the HPE CSI Driver for Kubernetes",
```
Comment (shivamerla) [inline operators/hpe-csi-operator/deploy/olm-catalog/hpe-csi-driver-operator/2.0.0/hpe-csi-driver-operator.v2.0.0.clusterserviceversion.yaml:35]: remove the comma for previous line.

File: docs/index.yaml (hunk 8 above)
Comment (bhagyashree-sarawate) [reply docs/index.yaml]: done

File: helm/charts/hpe-csi-driver/values.yaml [hunk 10]
```diff
@@ -9,7 +9,7 @@ imagePullPolicy: 'Always'
 backendType: nimble
 
 secret:
-  # parameters for nimble
+  # parameters for specified backendType (nimble, hpe3parprimera)
```
Comment (shivamerla) [inline helm/charts/hpe-csi-driver/values.yaml:12]: can you change this to 3parprimera too?

Comment (bhagyashree-sarawate) [reply helm/charts/hpe-csi-driver/values.yaml:12]: Modified to primera3par, since as per the kubernetes limitation kind name can not start with number

**Pull request merged**